frontend/components/player/player.jsx

```diff
@@ -23,16 +22,30 @@ const Player = ({ tracks, isPlaying, reduxPlay }) => {
 		audioRef.current.src = audioSrc;
 	}, [currentTrack]);
 
+	// Safely play audio only when it is loaded
+
+	const isReady = () => {
+		if (audioRef.current.readyState >= 2) {
+			audioRef.current.play();
+		}
+	};
+	const isReadyListener = () => {
+		audioRef.current.addEventListener("loadeddata", isReady);
+	};
+
 	// Set up play/pause behavior;
 	useEffect(() => {
 		if (isPlaying) {
-			setAfterFirstPlay(true);
+			isReadyListener();
 			audioRef.current.play();
```

Comment:
Removing this line leads to some instances when hitting the Play button on the ArtistShow page doesn't restart playback, though I'm not sure why.